extensions/positron-assistant/src/extension.ts

```diff
@@ -286,10 +287,18 @@ export function clearTokenUsage(context: vscode.ExtensionContext, provider: stri
 }
 
 // Registry to store token usage by request ID for individual requests
-const requestTokenUsage = new Map<string, { inputTokens: number; outputTokens: number }>();
+const requestTokenUsage = new Map<string, { inputTokens: number; outputTokens: number; provider: string }>();
 
-export function recordRequestTokenUsage(requestId: string, inputTokens: number, outputTokens: number) {
-	requestTokenUsage.set(requestId, { inputTokens, outputTokens });
+export function recordRequestTokenUsage(requestId: string, provider: string, inputTokens: number, outputTokens: number) {
+	const enabledProviders = vscode.workspace.getConfiguration('positron.assistant').get('approximateTokenCount', [] as string[]);
```

Comment:
Thoughts on adding this separate setting vs. reading the testModels and enabledProviders settings? If any of the experimental providers are included, we could show token usage as long as `showTokenUsage` is enabled?

Reply:
Oh, I also think the high number of tokens sent is what users are suspecting when using Assistant. I'm pretty sure it's the prompts and context that's inflating the tokens. It's part of the token counting from my understanding of what providers consider a token. It's probably the default prompt since it's quite lengthy.